pkg/scheduler/framework/types.go

```diff
@@ -947,3 +947,20 @@ func (h HostPortInfo) sanitize(ip, protocol *string) {
 		*protocol = string(v1.ProtocolTCP)
 	}
 }
+
+// InsufficientResource describes what kind of resource limit is hit and caused the pod to not fit the node.
+type InsufficientResource struct {
```

Comment:
This is not generic enough to be part of the framework. If anything, it can be within the `noderesources` package.

Reply:
`noderesources` doesn't seem a good fit, as eventually we will move it to component-helpers.

Let's comment out that this is a temporary location.

Reply:
I think `noderesources` is a better temporary location. It kind of prevents other plugins from using the type.

Reply:
SGTM. Let's revert to put it into noderesource - along with the new `NonResource` type.

Reply:
Okay, I tried to put `InsufficientResource` and `NonResource` into noderesources/fit.go.